include/openssl/ssl.h

```diff
@@ -250,6 +250,31 @@ typedef int (*tls_session_secret_cb_fn) (SSL *s, void *secret,
                                          STACK_OF(SSL_CIPHER) *peer_ciphers,
                                          const SSL_CIPHER **cipher, void *arg);
 
+/* Extension context codes */
+/* This extension is only allowed in TLS */
+#define SSL_EXT_TLS_ONLY                        0x0001
+/* This extension is only allowed in DTLS */
+#define SSL_EXT_DTLS_ONLY                       0x0002
+/* Some extensions may be allowed in DTLS but we don't implement them for it */
+#define SSL_EXT_TLS_IMPLEMENTATION_ONLY         0x0004
+/* Most extensions are not defined for SSLv3 but EXT_TYPE_renegotiate is */
+#define SSL_EXT_SSL3_ALLOWED                    0x0008
+/* Extension is only defined for TLS1.2 and below */
+#define SSL_EXT_TLS1_2_AND_BELOW_ONLY           0x0010
```

Comment:
I'm not sure I agree. Really SSL_EXT_SSL3_ALLOWED wouldn't normally be used. I've only included it for completeness because there is a built-in extension that uses it. There doesn't seems any great advantage for having separate flags for TLSv1.0, TLSv1.1 and TLSv1.2 - at least we haven't currently got any extensions that need that level of control. Splitting them up would mean you would need to specify 3 flags where currently you only need to specify 1.